catroid/src/main/java/org/catrobat/catroid/utils/FormulaEditorIntroUtil.java

```diff
@@ -0,0 +1,261 @@
+		}
+	}
+
+	public static void dismissIntro() {
+		if (isIntroVisible()) {
+			introWindow.dismiss();
+		}
+	}
+
+	public static void finishIntro() {
+		dismissIntro();
+		currentPage = 0;
+		prepared = false;
+		SnackbarUtil.setHintShown(activity, introId);
+	}
+
+	public static boolean isIntroVisible() {
+		return (introWindow != null && introWindow.isShowing());
+	}
+
+	private static void setHighlightParams(View fragmentView) {
+		int idx = 0;
+		setParamsOfView(null, idx++, 0, 0);
+		setParamsOfView((View) fragmentView.findViewById(R.id.formula_editor_edit_field), idx++, 1, 1);
+		setParamsOfView((View) fragmentView.findViewById(R.id.formula_editor_keyboard_7), idx++, 4, 4);
+		setParamsOfView((View) fragmentView.findViewById(R.id.formula_editor_keyboard_compute), idx++, 1, 1);
+		setParamsOfView((View) fragmentView.findViewById(R.id.formula_editor_keyboard_object), idx++, 1, 1);
+		setParamsOfView((View) fragmentView.findViewById(R.id.formula_editor_keyboard_function), idx++, 1, 1);
+		setParamsOfView((View) fragmentView.findViewById(R.id.formula_editor_keyboard_logic), idx++, 1, 1);
+		setParamsOfView((View) fragmentView.findViewById(R.id.formula_editor_keyboard_sensors), idx++, 1, 1);
+		setParamsOfView((View) fragmentView.findViewById(R.id.formula_editor_keyboard_data), idx++, 1, 1);
+	}
+
+	private static void setIntroStrings() {
+		int idx = 0;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_formula_editor;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_input_field;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_keyboard;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_compute;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_object;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_functions;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_logic;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_device;
+		titleStringIds[idx++] = R.string.formula_editor_intro_title_data;
+		idx = 0;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_formula_editor;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_input_field;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_keyboard;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_compute;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_object;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_functions;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_logic;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_device;
+		summaryStringIds[idx++] = R.string.formula_editor_intro_summary_data;
+	}
+
+	private static void setParamsOfView(View view, int highlightRectangleIndex, int widthModifier, int heightModifier) {
+		int width = 1;
+		int height = 1;
+		int[] coords = new int[2];
+		coords[0] = 0;
+		coords[1] = 0;
+
+		if (view != null) {
```

Comment:
yes. See line '157' I pass it as an argument because the first page of the intro does not have a hightlight area.